internal/pkg/template/mappers.go

```diff
@@ -74,14 +72,22 @@ func MappersFor(s *state.State, d dependencies, ctx Context) (mapper.Mappers, er
 		replacevalues.NewMapper(s, replacements),
 	}
 
+	// Add metadata on "template preview" operation
+	if c, ok := ctx.(previewContext); ok {
+		if _, isUseContext := ctx.(useContext); !isUseContext {
+			mappers = append(mappers, metadata.NewMapperWithoutInstanceID(
+				s,
+				c.TemplateRef(),
+				c.ObjectIds(),
+				c.InputsUsage(),
+			))
+		}
+	}
+
```

Comment:
The logic for adding metadata mappers in the template preview operation was previously duplicated.  This update refactors the code to ensure that the metadata. NewMapperWithoutInstanceID mapper is added only once for the template preview operation.